evalml/utils/woodwork_utils.py

```diff
@@ -0,0 +1,112 @@
+
+import numpy as np
+import pandas as pd
+import woodwork as ww
+
+numeric_and_boolean_ww = [ww.logical_types.Integer, ww.logical_types.Double, ww.logical_types.Boolean]
+
+
+def infer_feature_types(data, feature_types=None):
+    """Create a Woodwork structure from the given pandas or numpy input, with specified types for columns.
+        If a column's type is not specified, it will be inferred by Woodwork.
+
+    Arguments:
+        data (pd.DataFrame): Input data to convert to a Woodwork data structure.
+        feature_types (string, ww.logical_type obj, dict, optional): If data is a 2D structure, feature_types must be a dictionary
+            mapping column names to the type of data represented in the column. If data is a 1D structure, then feature_types must be
+            a Woodwork logical type or a string representing a Woodwork logical type ("Double", "Integer", "Boolean", "Categorical", "Datetime", "NaturalLanguage")
+
+    Returns:
+        A Woodwork data structure where the data type of each column was either specified or inferred.
+    """
+    ww_data = _convert_to_woodwork_structure(data)
+    if feature_types is not None:
+        if len(ww_data.shape) == 1:
+            ww_data = ww_data.set_logical_type(feature_types)
+        else:
+            ww_data = ww_data.set_types(logical_types=feature_types)
```

Comment:
This function would probably be more efficient if you could pass in any specified logical types to the woodwork initialization call in `_convert_to_woodwork_structure`. With the current approach, you end up inferring the types for all columns and then possibly changing them. If you could pass any known logical types to the woodwork init call, you could avoid the inference cost for those columns.

Reply:
Yes, you're totally right and this falls within the discussion we had about updating custom types. Going to remove `_convert_to_woodwork_structure` and move the impl to `infer_feature_types` since they're basically the same.